Update Model Builder Doc (#24909)
dotnet/docs: docs/machine-learning/automate-training-with-model-builder.md

You don't need machine learning expertise to use Model Builder. All you need is some data, and a problem to solve. Model Builder generates the code to add the model to your .NET application.

> ![Model Builder Scenarios](./media/model-builder-scenarios.png)

Reviewer: @briacht what do you think about the screenshot? Should we zoom out a bit to show MB is part of Visual Studio?

Reviewer 2: Yeah I think showing all of Visual Studio would be good

> [!NOTE]
> Model Builder is currently in Preview.

> ## Creating a Model Builder Project

Reviewer: I would expand on this a bit with this information

https://devblogs.microsoft.com/dotnet/ml-net-and-model-builder-march-updates/

```markdown
After training, two files are generated under the *.mbconfig file:

- Model.consumption.cs: This file contains the Model Input and Model Output schemas as well as the Predict function generated for consuming the model.
- Model.training.cs: This file contains the training pipeline (data transforms, algorithm, algorithm hyperparameters) chosen by Model Builder to train the model. You can use this pipeline for re-training your model.
- Model.zip: This is a serialized zip file which represents your trained ML.NET model.
```

> When you first start up Model Builder it will ask for you to name the project. This will create an `mbconfig` file inside of the project.

Reviewer: ```suggestion
When you first start up Model Builder it will ask for you to name the project. This will create an `mbconfig` configuration file inside of the project.
```

You don't need machine learning expertise to use Model Builder. All you need is some data, and a problem to solve. Model Builder generates the code to add the model to your .NET application.

![Model Builder Scenarios](./media/model-builder-scenarios.png#lightbox)

> [!NOTE]
> Model Builder is currently in Preview.

## Creating a Model Builder Project

When you first start up Model Builder it will ask for you to name the project. This will create an `mbconfig` configuration file inside of the project.

The `mbconfig` file keeps track of everything you do in Model Builder to allow you to reopen the session.

> After training, two files are generated under the *.mbconfig file:

Reviewer: ```suggestion
After training, three files are generated under the *.mbconfig file:
```

> 

Reviewer: Add a note.

When you create your `mbconfig` file, you're prompted for a name. This name is applied to the consumption, training, and model files. In this case, the name used is *Model*.

> - Model.consumption.cs: This file contains the Model Input and Model Output schemas as well as the Predict function generated for consuming the model.

Reviewer: ```suggestion
- **Model.consumption.cs:** This file contains the `ModelInput` and `ModelOutput` schemas as well as the `Predict` function generated for consuming the model.
```

> - Model.training.cs: This file contains the training pipeline (data transforms, algorithm, algorithm hyperparameters) chosen by Model Builder to train the model. You can use this pipeline for re-training your model.

Reviewer: ```suggestion
- **Model.training.cs:** This file contains the training pipeline (data transforms, algorithm, algorithm hyperparameters) chosen by Model Builder to train the model. You can use this pipeline for re-training your model.
```

> - Model.zip: This is a serialized zip file which represents your trained ML.NET model.

Reviewer: ```suggestion
- **Model.zip:** This is a serialized zip file which represents your trained ML.NET model.
```

![Diagram showing Model Builder steps](media/model-builder-steps.png)

> Model Builder supports datasets in .tsv, .csv, .txt formats, as well as SQL database format. If you have a .txt file, columns should be separated with `,`, `;` or `/t`.

Reviewer: Is the delimiter correct? I think it's `\`

```suggestion
Model Builder supports datasets in .tsv, .csv, .txt formats, as well as SQL database format. If you have a .txt file, columns should be separated with `,`, `;` or `\t`.
```

Author: Good catch!

- Balance your data. For classification tasks, make sure that the training set is balanced across the categories. For example, if you have four classes for 100 training examples, and the two first classes (tag1 and tag2) are used for 90 records, but the other two (tag3 and tag4) are only used on the remaining 10 records, the lack of balanced data may cause your model to struggle to correctly predict tag3 or tag4.

## Consume

After the evaluation phase, Model Builder outputs a model file, and code that you can use to add the model to your application. ML.NET models are saved as a zip file. The code to load and use your model is added as a new project in your solution. Model Builder also adds a sample console app that you can run to see your model in action.

In addition, Model Builder gives you the option to create projects that consume your model. Currently, Model Builder will create the following projects:

- Console app - Creates a .NET Core console applications to make predictions from your model.

> - Web API - Creates an ASP.NET Core Web API that consumes your model.

Reviewer: ```suggestion
- Web API - Creates an ASP.NET Core Web API that lets you consume your model over the internet.
```

- Balance your data. For classification tasks, make sure that the training set is balanced across the categories. For example, if you have four classes for 100 training examples, and the two first classes (tag1 and tag2) are used for 90 records, but the other two (tag3 and tag4) are only used on the remaining 10 records, the lack of balanced data may cause your model to struggle to correctly predict tag3 or tag4.

## Consume

After the evaluation phase, Model Builder outputs a model file, and code that you can use to add the model to your application. ML.NET models are saved as a zip file. The code to load and use your model is added as a new project in your solution. Model Builder also adds a sample console app that you can run to see your model in action.

In addition, Model Builder gives you the option to create projects that consume your model. Currently, Model Builder will create the following projects:

> - Console app - Creates a .NET Core console applications to make predictions from your model.
> - Web API - Creates an ASP.NET Core Web API that lets you consume your model over the internet.

Reviewer: ```suggestion
- **Console app:** Creates a .NET Core console applications to make predictions from your model.
- **Web API:** Creates an ASP.NET Core Web API that lets you consume your model over the internet.
```
